### DISCUSSION TITLE
Adding Modded Tools For Turtles

### DISCUSSION THREAD

**Discussion by XdecoraX** [cc-tweaked/CC-Tweaked#1966] [Q&A] [answered]
I’ve looked through some of the previous posts about creating custom turtle tools but wasn’t able to find anything that helped. I’ve also looked at the CC Tweaked datapack builder and tried to use that as an example to build off of. I’m not sure what I need to be changing in the datapack to have it add the tool. I figured it would be “namespace/data/modid/computercraft/turtle_upgrades/tool_name”
And then the only thing I would need to change in the json file would be the “item”: “item name”.

**Comment by SquidDev** [comment]
> I figured it would be `namespace/data/modid/computercraft/turtle_upgrades/tool_name`
> And then the only thing I would need to change in the json file would be the `"item": "item name"`.

It should be `tool_name.json`, but otherwise yes. Is this not working — are there any errors in your log file?

**Reply by XdecoraX** [reply]
Didn't think about checking the logs to see if they were loading properly, but upon checking they are not loading correctly and are throwing the following error.
 
![Capture](https://github.com/user-attachments/assets/b025a27e-912c-4ceb-a5c2-47a9a9caeb8c)

I’m assuming that I’m doing something wrong as I’ve included the netherite tools that the CC Tweaked datapack builder provides and that is correctly loading. I’ve attached what my datapack currently looks like. I’m currently trying to add it for railcraft as a test but I’m ultimately trying to use it on another mod that has their tools structured differently than the “normal”.
[CC-Custom.zip](https://github.com/user-attachments/files/16841652/CC-Custom.zip)

**Reply by SquidDev** [reply, marked as answer]
Ahh, you need to use the fully-namespaced name. So `"item": "railcraft:steel_axe"`

**Reply by XdecoraX** [reply]
That was my issue! Thank you so much for the help!
